app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx

```diff
@@ -10,20 +10,107 @@ import * as loggerProvider from "../providers/logger";
 // === Constants ===
 // =================
 
+/** Default HTTP body for an "open project" request. */
+const DEFAULT_OPEN_PROJECT_BODY: OpenProjectRequestBody = {
+  forceCreate: false,
+};
+
 /** Relative HTTP path to the "get user" endpoint of the Cloud backend API. */
 const GET_USER_PATH = "users/me";
+/** Relative HTTP path to the "list projects" endpoint of the Cloud backend API. */
+const LIST_PROJECTS_PATH = "projects";
+/** Relative HTTP path to the "create project" endpoint of the Cloud backend API. */
+const CREATE_PROJECT_PATH = "projects";
+/** Relative HTTP path to the "close project" endpoint of the Cloud backend API. */
+const closeProjectPath = (projectId: ProjectId) =>
+  `projects/${projectId}/close`;
+/** Relative HTTP path to the "get project" endpoint of the Cloud backend API. */
+const getProjectPath = (projectId: ProjectId) => `projects/${projectId}`;
+/** Relative HTTP path to the "open project" endpoint of the Cloud backend API. */
+const openProjectPath = (projectId: ProjectId) => `projects/${projectId}/open`;
 
 // =============
 // === Types ===
 // =============
 
+/** Unique identifier for a user's project. */
+export type ProjectId = string;
+
 /** A user/organization in the application. These are the primary owners of a project. */
 export interface Organization {
   id: string;
   userEmail: string;
   name: string;
 }
 
+/** Type of application that a {@link Version} applies to.
+ *
+ * We keep track of both backend and IDE versions, so that we can update the two independently.
+ * However the format of the version numbers is the same for both, so we can use the same type for
+ * both. We just need this enum to disambiguate. */
+export enum VersionType {
+  backend = "Backend",
+  ide = "Ide",
+}
+
+/** A version describing a release of the backend or IDE. */
+export interface Version {
+  versionType: VersionType;
+  ami: string | undefined;
+  created: string;
+  /** This field name does not follow the naming convention. This field name is snake case on the
+   * backend. The case must match here for JSON deserialization to work. */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  version_number: string;
+}
+
+/** Possible states that a project can be in. */
+export enum ProjectState {
+  created = "Created",
+  new = "New",
+  openInProgress = "OpenInProgress",
+  opened = "Opened",
+  closed = "Closed",
+}
+
+/** Wrapper around a project state value. */
+export interface ProjectStateType {
+  type: ProjectState;
+}
+
+/** A user/organization's project containing and/or currently executing code. */
+export interface Project {
+  organizationId: string;
+  projectId: ProjectId;
+  name: string;
+  state: ProjectStateType;
+  packageName: string;
+  address: string | null;
+  ami: string | null;
+  ideVersion: Version | null;
```

Comment:
types may be incorrect, the types from the backend look like this, it might be wrong if serde is doing something funky when serializing though:
https://github.com/enso-org/enso/blob/5fb66bb0679db2a62b8f43398e565f2e2e708ba9/app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx#L155-L156

Reply:
Dashboard backend methods extracted to their own PR, so this is no longer relevant.